addon-test-support/@ember/test-helpers/dom/double-click.js

```diff
@@ -8,20 +8,21 @@ import { nextTickPromise } from '../-utils';
 /**
   @private
   @param {Element} element the element to double-click on
+  @param {Object} options the options to be merged into the mouse events
 */
-export function __doubleClick__(element) {
-  fireEvent(element, 'mousedown');
+export function __doubleClick__(element, options) {
+  fireEvent(element, 'mousedown', options);
 
   if (isFocusable(element)) {
     __focus__(element);
   }
 
-  fireEvent(element, 'mouseup');
-  fireEvent(element, 'click');
-  fireEvent(element, 'mousedown');
-  fireEvent(element, 'mouseup');
-  fireEvent(element, 'click');
-  fireEvent(element, 'dblclick');
+  fireEvent(element, 'mouseup', options);
+  fireEvent(element, 'click', options);
+  fireEvent(element, 'mousedown', options);
+  fireEvent(element, 'mouseup', options);
+  fireEvent(element, 'click', options);
+  fireEvent(element, 'dblclick', options);
```

Comment:
Same concern here. If we are going to allow event options I think we need a system that allows us to customize each of the events individually...

@mmun and I had discussed this somewhat recently, but I can’t recall if we came to a good conclusion 🤔

Reply:
Maybe allow `options` to either be a POJO or a function, and if a function, gets invoked with the name of the event and its return value is the options POJO?